### PR TITLE
feat: add auth options for the api plugin capability

### PR DESCRIPTION
**Pull request by huimiu**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/27822940
This PR is for UI part, template code PR is [here](https://github.com/OfficeDev/teams-toolkit/pull/11747)

### CODE REVIEW

**Review by LongOddCode — APPROVED** [review]

**Review by hund030 — APPROVED** [review]

File: packages/fx-core/src/question/create.ts [hunk 1]
```diff
@@ -1091,18 +1093,26 @@ export function apiSpecLocationQuestion(includeExistingAPIs = true): SingleFileO
...
-    staticOptions: ApiMessageExtensionAuthOptions.all(),
-    dynamicOptions: () => ApiMessageExtensionAuthOptions.all(),
-    default: ApiMessageExtensionAuthOptions.none().id,
+    staticOptions: ApiAuthOptions.all(),
+    dynamicOptions: (inputs: Inputs) => {
+      const options: OptionItem[] = [ApiAuthOptions.none()];
+      if (inputs[QuestionNames.MeArchitectureType] == MeArchitectureOptions.newApi().id) {
+        options.push(ApiAuthOptions.apiKey(), ApiAuthOptions.microsoftEntra());
+      } else if (inputs[QuestionNames.Capabilities] == CapabilityOptions.copilotPluginNewApi().id) {
+        options.push(ApiAuthOptions.apiKey(), ApiAuthOptions.oauth());
```
Comment (yuqizhou77) [inline packages/fx-core/src/question/create.ts:1111]: Do we need feature for these two options? We have "isCopilotAuthEnabled" for Copilot plugin with auth from existing API. @SLdragon please correct me if I am wrong.

File: packages/fx-core/src/question/create.ts [hunk 2]
```diff
@@ -1091,18 +1093,26 @@ export function apiSpecLocationQuestion(includeExistingAPIs = true): SingleFileO
...
       "core.createProjectQuestion.apiMessageExtensionAuth.placeholder"
     ),
     cliDescription: "The authentication type for the API.",
-    staticOptions: ApiMessageExtensionAuthOptions.all(),
-    dynamicOptions: () => ApiMessageExtensionAuthOptions.all(),
-    default: ApiMessageExtensionAuthOptions.none().id,
+    staticOptions: ApiAuthOptions.all(),
+    dynamicOptions: (inputs: Inputs) => {
+      const options: OptionItem[] = [ApiAuthOptions.none()];
+      if (inputs[QuestionNames.MeArchitectureType] == MeArchitectureOptions.newApi().id) {
```
Comment (yuqizhou77) [inline packages/fx-core/src/question/create.ts:1108]: ===. Same for below.

Comment (huimiu) [reply packages/fx-core/src/question/create.ts:1108]: Since the API ME authentication is GA, I think we don't need to add a feature flag for it

Comment (yuqizhou77) [reply packages/fx-core/src/question/create.ts:1108]: Oh I mean if we should use "===" rather than "=="

Comment (huimiu) [reply packages/fx-core/src/question/create.ts:1108]: I see, good catch, let me update in next PR

File: packages/fx-core/src/question/create.ts (hunk 1 above)
Comment (huimiu) [reply packages/fx-core/src/question/create.ts:1111]: Let me add the `isCopilotAuthEnabled` feature flag in next PR, thanks!

**Pull request merged**